lib/charms/bind/v0/dns_record.py

```diff
@@ -238,15 +240,53 @@ class RequirerDomain(BaseModel):
     Attributes:
         domain: the domain name.
         username: username for authentication.
+        password: the password for authentication.
         password_id: secret password for authentication.
         uuid: UUID for this entry.
     """
 
     domain: str = Field(min_length=1)
     username: str
+    password: Optional[str] = Field(default=None, exclude=True)
     password_id: str
     uuid: UUID
 
+    def get_password(self, model: ops.Model) -> str:
+        """Retrieve the password corresponding to the password_id.
+
+        Args:
+            model: the Juju model.
+
+        Returns:
+            the plain password.
+        """
+        secret = model.get_secret(id=self.password_id)
```

Comment:
Should it check that there is secret support ?

Reply:
Juju versions not supporting secrets won't be able to run the charms. We can set the requirement in the metadata, so I don't see why we'd like to do that here too

Reply:
That's an alternative, so either define the juju version, or check secret support